rbac/management/principal/cleaner.py

```diff
@@ -212,7 +213,14 @@ def process_umb_event(frame, umb_client: Stomp, bootstrap_service: TenantBootstr
             # If the setting is enabled, process all users.
             if not user.is_active or settings.PRINCIPAL_CLEANUP_UPDATE_ENABLED_UMB:
                 # If Tenant is not already ready, don't ready it
-                bootstrap_service.update_user(user, ready_tenant=False)
+                try:
+                    bootstrap_service.update_user(user, ready_tenant=False)
+                except ValueError as e:
+                    logger.error("process_umb_event: Error updating user: %s", str(e))
+                    capture_exception(e)
+                    umb_client.nack(frame)
+                    umb_message_processed_count.inc()
```

Comment:
I wonder if this may be misleading, since the message isn't actually "processed" in the "done" sense. Maybe we should add a label to the counter for this? I'm not sure what the best name. Maybe `frame="ack"` vs `frame="nack"`?

Reply:
Another option is to use a different metric.